nixos/modules/services/x11/desktop-managers/gnome3.xml

```diff
@@ -0,0 +1,246 @@
+</programlisting>
+
+   <para>
+    and none will be installed.
+   </para>
+
+   <para>
+    If you’d only like certain omissions you can use <xref linkend="opt-environment.gnome3.excludePackages"/>.
+    Note, this only removes things installed in the gnome3 module from <xref linkend="opt-environment.systemPackages"/>.
+    Certain applications require an entire nixos module, like <literal>evince</literal> or <literal>geary</literal>.
+    Search for GNOME apps under the <literal>programs.*</literal> namespace with the
+    <link xlink:href="https://nixos.org/nixos/options.html#">NixOS options search</link> to find which option to disable.
+   </para>
+  </section>
+
+  <section xml:id="sec-gnome-games">
+   <title>GNOME games</title>
+
+   <para>
+    You can install all of the GNOME games with:
+   </para>
+
+<programlisting>
+services.gnome3.games.enable = true;
+</programlisting>
+  </section>
+ </section>
+
+ <section xml:id="sec-gnome-enable-flashback">
+  <title>Enabling GNOME Flashback</title>
+
+  <para>
+   You can enable the default GNOME Flashback session, which uses metacity, with:
+  </para>
+
+<programlisting>
+<xref linkend="opt-services.xserver.desktopManager.gnome3.flashback.enableMetacity"/> = true;
+</programlisting>
+
+  <para>
+   It’s also possible to create custom sessions with <xref linkend="opt-services.xserver.desktopManager.gnome3.flashback.customSessions"/>.
+  </para>
+
+  <para>
+   The following example uses <literal>xmonad</literal>:
+  </para>
+
+<programlisting>
+<xref linkend="opt-services.xserver.desktopManager.gnome3.flashback.customSessions"/> = [{
+  wmName = &quot;xmonad&quot;;
+  wmLabel = &quot;XMonad&quot;;
+  wmCommand = &quot;${pkgs.haskellPackages.xmonad}/bin/xmonad&quot;;
+}];
+</programlisting>
+
+ </section>
+ <section xml:id="sec-gnome-gdm">
+  <title>GDM</title>
+
+  <para>
+   If you want to use GNOME wayland and NVidia you need to enable:
+  </para>
+
+<programlisting>
+<xref linkend="opt-services.xserver.displayManager.gdm.nvidiaWayland"/> = true;
+</programlisting>
+
+  <para>
+   as the default configuration will forbid this.
+  </para>
+ </section>
+
+ <section xml:id="sec-gnome-icons-and-gtk-themes">
+  <title>Icons and GTK Themes</title>
+
+  <para>
+   Icon themes and GTK themes don’t require any special option to install in NixOS.
+  </para>
+
+  <para>
+   You can add them to <xref linkend="opt-environment.systemPackages"/> and switch to them with GNOME Tweaks.
+   If you’d like to do this manually in dconf, change the values of the following keys:
+  </para>
+
+<programlisting>
+/org/gnome/desktop/interface/gtk-theme
+/org/gnome/desktop/interface/icon-theme
+</programlisting>
+
+  <para>
+   in <literal>dconf-editor</literal>
+  </para>
+ </section>
+
+ <section xml:id="sec-gnome-shell-extensions">
+  <title>Shell Extensions</title>
+
+  <para>
+   All shell extensions are packaged under the <literal>gnomeExtensions</literal> attribute.
```

Comment:
:woman_shrugging: Some packages elsewhere also contain extensions as their part (`gnome3.gpaste` comes to mind).